src/main/connect/connectionManager.ts

```diff
@@ -184,32 +177,34 @@ export class ConnectionManager implements ExtensionComponent, vscode.Disposable
     public areCompleteCredentialsSet(): boolean {
         return !!(this._xrayUrl && this._rtUrl && ((this._username && this._password) || this._accessToken));
     }
-
-    public async verifyCredentials(prompt: boolean): Promise<boolean> {
-        if (!this.areXrayCredentialsSet()) {
-            return false;
+    /**
+     * Test whether the connection is valid and the user has permissions to access Xray and Artifactory.
+     */
+    public async testConnection(prompt: boolean): Promise<TestConnectionStatus> {
+        if (!(await this.testConnectionHealth())) {
+            this.deleteCredentialsFromMemory();
+            return TestConnectionStatus.FailedServerNotFound;
```

Comment:
```suggestion
```
And what if we get 401 in the ping? This request seems redundant, because testComponentPermission already handle the response parsing.